src/main/java/org/apache/commons/configuration2/ImmutableConfiguration.java

```diff
@@ -62,6 +62,20 @@ public interface ImmutableConfiguration {
      */
     boolean containsKey(String key);
 
+    /**
+     * Returns true if this configuration contains one or more matches to this value. This operation stops at first
+     * match but may be more expensive than the {@link #containsKey containsKey} method.
+     *
+     * @param value value whose presence in this configuration is to be tested
+     * @return {@code true} if this configuration maps one or more keys to the specified value
+     * @throws NullPointerException          if the value is {@code null}
+     * @throws UnsupportedOperationException if this method is not supported by the implementing class.
+     */
+    default boolean containsValue(String value) {
+        throw new UnsupportedOperationException(
+            String.format("Class %s does not support this method.", this.getClass()));
```

Comment:
The message is redundant and bad, it read like "Class class com.example.Foo does not...".
You must have not tested this case. A simpler message could just say `getClass().getName() + ".myMethod"`

Reply:
Hey Greg, you are right

I did in fact not test this method since I couldn't think of a way to test this in a class that doesn't implement the method without creating a dummy class for it.

Do you think I should just adapt the message like you suggested?

Thank you.

Reply:
Yes.